### ISSUE TITLE
Knee phase angle not calibrated  

### ISSUE THREAD

**Issue by AndPatr** [ADVRHumanoids/awesome_leg#16]
Continuation of the discussion on the phase angle problem started in [torque_read_out_of_range_issue](https://github.com/ADVRHumanoids/awesome_leg/issues/12).
@ntsagarakis, @stecorda

**Comment by AndPatr** [comment]
So - to answer the last question @stecorda made -  if you have time the maintenance on the knee joint could be done even this afternoon. I just have to make a quick test for checking the current dampening sheet configuration and that you can work on the leg. 
@ntsagarakis, since we have to dismount the joint I propose we also check the hip joint phase angle calibration and seize the opportunity to check the high friction we have been experiencing from the start on the knee (maybe is something we can solve easily).
@stecorda if you want I can be there while you check the knee.  Also remember that yesterday we felt a smell of burned plastic from the knee. so maybe there is a plastic component which melted down

**Comment by AndPatr** [comment]
Now, for what concerns the results of the test of yesterday evening (video at [cyclic_jump_test](https://istitutoitalianotecnologia-my.sharepoint.com/:v:/r/personal/andrea_patrizi_iit_it/Documents/20221107_184005_1.mp4?csf=1&web=1&e=d01pVw) ).
I'll put here some results of the post-processing, since they might be useful to shed light on possible hardware issues.

During the test, the leg jumped 14 times without evident issues (the decision to stop the test to this number of jumps was simply my call, but we could have gone for longer).

This is the impedance setpoint during the tests:
![Figure_10](https://user-images.githubusercontent.com/90312267/200578033-a13ae2c8-2403-4e91-b3dc-a12ca07ef000.png)

These are the measured efforts:
![Figure_5](https://user-images.githubusercontent.com/90312267/200578185-4d29a063-1aa5-4a79-a6bc-aae8c0145c50.png)
During the last 5 jumps I completely removed the filters to check that this worked on the leg not only is simulation and, I don't really know why as of now, the measured peak torques are lower.

These are the quadrature currents:
![Figure_1](https://user-images.githubusercontent.com/90312267/200578741-7b361b65-1573-4539-92c7-7b465360337e.png)
Driver temperature:
![Figure_6](https://user-images.githubusercontent.com/90312267/200579275-5a967819-dcce-4979-80fe-f952f56b62bf.png)
As a side note, these are the measured VS estimated quadrature currents:
![Figure_9](https://user-images.githubusercontent.com/90312267/200579667-d5ad8375-6f9e-4649-b4f9-616f8963f31a.png)

**Comment by ntsagarakis** [comment]
> So - to answer the last question @stecorda made - if you have time the maintenance on the knee joint could be done even this afternoon. I just have to make a quick test for checking the current dampening sheet configuration and that you can work on the leg.
> @ntsagarakis, since we have to dismount the joint I propose we also check the hip joint phase angle calibration and seize the opportunity to check the high friction we have been experiencing from the start on the knee (maybe is something we can solve easily).
> @stecorda if you want I can be there while you check the knee. Also remember that yesterday we felt a smell of burned plastic from the knee. so maybe there is a melted plastic component somewhere.

We can do the full maintenance for the phase angles but for the friction we will need to disassemble the drive probably that will take time, not sure if we can have it ready for the review then next week

**Comment by ntsagarakis** [comment]
>>During the last 5 jumps I completely removed the filters to check that this worked on the leg not only is simulation and, I >>don't really know why as of now, the measured peak torques are lower.

Strange, did motion performance changed, e.g. jumping height when filters were removed?  if not as also there is not an effect on the current probably this is a filter gain issue somewhere

**Comment by alaurenzi** [comment]
@ntsagarakis @stecorda it looks like the updated FW is saturating the current to 30 A regardless of the ecat master configuration. For this prototype, we need way more (ideally, whatever won't break the driver) !

**Comment by stecorda** [comment]
> So - to answer the last question @stecorda made - if you have time the maintenance on the knee joint could be done even this afternoon. I just have to make a quick test for checking the current dampening sheet configuration and that you can work on the leg. @ntsagarakis, since we have to dismount the joint I propose we also check the hip joint phase angle calibration and seize the opportunity to check the high friction we have been experiencing from the start on the knee (maybe is something we can solve easily). @stecorda if you want I can be there while you check the knee. Also remember that yesterday we felt a smell of burned plastic from the knee. so maybe there is a melted plastic component somewhere.

Today I had a meeting in Morego. Tomorrow I'm not yet sure I'll be in iit so thursday I will check it. Leave both actuators on my desk so I can start to test them in the early morning.

**Comment by stecorda** [comment]
> @ntsagarakis @stecorda it looks like the updated FW is saturating the current to 30 A regardless of the ecat master configuration. For this prototype, we need way more (ideally, whatever won't break the driver) !

The fw is not changed since yesterday morning where we've seen 50 amps also for a long time then I don't think is a fw problem, for the rest we can try to find a spare pair of 100A boards but this level of current can issued only for a very limited time, not seconds

**Comment by AndPatr** [comment]
> The fw is not changed since yesterday morning where we've seen 50 amps also for a long time then I don't think is a fw problem, for the rest we can try to find a spare pair of 100A boards but this level of current can issued only for a very limited time, not seconds
That actually true. I just checked from the gui and I was able to get current references beyond 30 A.
I will check again by logging data from a new jump session, but the current limit is and was set to 55A.

**Comment by ntsagarakis** [comment]
Stefano,  what is setting the limit apart from the thermal, is it the current sensor measurement range? We not need to go above 60A. what is the current sensor measurement range on these boards that are on the joints now?

**Comment by AndPatr** [comment]
Update on the current saturation "issue".
Yesterday I rerun a jump trial and this morning I looked at the data to see if we were still getting saturated iq at 30 A.
I have no clue as to why now we are using the correct current limits:
![Figure_1](https://user-images.githubusercontent.com/90312267/200798826-b22e1ec8-60f6-4cee-9e22-8f7618ac811a.png)
I have checked the commits on the ecat master config file and the current limit was set to 55A since the 10th of October, so I don't know how it is possible that we saw saturation to 30A. 
I'll leave the investigation as to why this happened in background for now.
So @ntsagarakis, @stecorda all that is left is to re-calibrate the magnetic angle offset of the leg and clean the surface of the ground under the test rig.
